enhancements/network/ovn-on-hypershift.md

```diff
@@ -0,0 +1,275 @@
+
+If the bandwidth or performance is not acceptable, will we have to deploy some kind of ovsdb proxy on the infra nodes? Assuming we cannot *require* infra nodes, will we nevertheless need to make that option available to customers?
+
+How do we report control-plane status -- which references "invisible" pods -- to the end-user? Does the end-user need to know that ovsdb is somehow degraded? What do we show in the `network` ClusterOperator object?
+
+How do we report control-plane status to the SRE teams? They will need to react to losses of quorum, etc. Can we deploy alerts? How do we handle cross-version alert definitions?
+
+How do we detect guest cluster MTU? We cannot just probe from the CNO as normal. We will either need to require MTU to be configured, or execute a probe pod. We do, however, support changing MTU
+within the CNO now, so it may be sufficient to default to 1500 and allow increases.
+
+### Test Plan
+
+The general e2e test suites will be acceptable.  We expect OVN to be the default for all hypershift based OpenShift clusters once OVN support is ready, so anywhere we are testing hypershift will be exercising this integration.
+
+Additionally, we will need to write some disruption tests that ensure ovsdb raft plays well with `StatefulSet` rotations.
+
+We will work closely with the Performance & Scale test team to test out the performance of this architecture.
+
+### Graduation Criteria
+
+Once this functionality is complete, we expect to make OVN the default for hypershift managed clusters. At that point, the graduation of this functionality will be tied to the graduation of hypershift more broadly.
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+#### Removing a deprecated feature
+n/a
+### Upgrade / Downgrade Strategy
+
+This architecture will only apply to new clusters deployed via hypershift. Thus, the generalized Hypershift upgrade model must take OVN in to account.
+
+OVN has somewhat complicated upgrade ordering. On upgrades, nodes must be upgraded before the control plane. For downgrades, the control plane must be rolled out first. While this logic is unchanged in Hypershift, splitting the components across clusters makes this more complicated w.r.t. potential failure cases.
```

Comment:
The CNO always go first, and will manage the upgade ordering. It should Just Work, and we've not had issues in the field so far. Just something to call out.

Reply:
If there is a requirement to upgrade control plane first,  then it can be done.  OVN supports the option to PIN ovn-controller and ovn-northd to a specific version.  More details here - https://github.com/ovn-org/ovn/commit/1dd27ea7aea40122c1edbff845e14abaa70c0413

CNO needs to be enhanced if OCP/Hypershift wants to make use of this.